pkg/proxy/apis/config/validation/validation.go

```diff
@@ -351,3 +300,55 @@ func validateInterface(iface string, fldPath *field.Path) field.ErrorList {
 	}
 	return allErrs
 }
+
+func validateMasqueradeBit(masqueradeBit *int32, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if masqueradeBit != nil && (*masqueradeBit < 0 || *masqueradeBit > 31) {
+		allErrs = append(allErrs, field.Invalid(fldPath, masqueradeBit, "must be within the range [0, 31]"))
+	}
+	return allErrs
+}
+
+func validateDualStackCIDRStrings(cidrs []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	switch {
+	case len(cidrs) == 0:
+		allErrs = append(allErrs, field.Invalid(fldPath, cidrs, "must contain at least one CIDR"))
+	case len(cidrs) > 2:
+		allErrs = append(allErrs, field.Invalid(fldPath, cidrs, "must be a valid CIDR list (e.g. [10.100.0.0/16, fde4:8dba:82e1::/48]"))
```

Comment:
a list of more than 2 CIDRs is a "valid CIDR list", it's just not allowed in this context. you need to say "must be either a single CIDR or a dual-stack pair of CIDRs" or somethign like that